Source/Core/Core/IOS/DI/DI.cpp

```diff
@@ -57,13 +57,13 @@ void DIDevice::DoState(PointerWrap& p)
   p.Do(m_last_length);
 }
 
-IPCCommandResult DIDevice::Open(const OpenRequest& request)
+std::optional<IPCReply> DIDevice::Open(const OpenRequest& request)
```

Comment:
Quick guide to reviewing:

* command handlers return `std::optional<IPCReply>` or `IPCReply`
* IPCCommandResult -> IPCReply
* GetNoReply -> std::nullopt
* GetDefaultReply -> IPCReply